api/v1beta1/dbinstance_types.go

```diff
@@ -72,15 +72,25 @@ type BackendServer struct {
 // and describes necessary informations to use instance
 // generic instance can be any backend, it must be reachable by described address and port
 type GenericInstance struct {
-	Host     string `json:"host"`
-	Port     uint16 `json:"port"`
-	PublicIP string `json:"publicIp,omitempty"`
+	Host         string   `json:"host,omitempty"`
+	HostFrom     *FromRef `json:"hostFrom,omitempty"`
+	Port         uint16   `json:"port,omitempty"`
+	PortFrom     *FromRef `json:"portFrom,omitempty"`
+	PublicIP     string   `json:"publicIp,omitempty"`
+	PublicIPFrom *FromRef `json:"publicIpFrom,omitempty"`
```

Comment:
Should we also take a connection url as a possible input?

Reply:
Do we have it now? I've just added From to all the fields that are there now, except the BackupHost because we'll replace it soon